N_Queen.java

```diff
@@ -0,0 +1,84 @@
+// Time Complexity : O(N!) n factorial. At each cell we have n(n-1) cells to choose.
```

Comment:
time complexity is n^n not n!

Reply:
The time is n!. We discussed same in our December cohort as well
https://leetcode.com/problems/n-queens/solution/